docs/guides/src/main/server/configuration_overview.adoc

```diff
@@ -0,0 +1,177 @@
+<#import "/templates/guide.adoc" as tmpl>
+<#import "/templates/kc.adoc" as kc>
+<#import "/templates/options.adoc" as opts>
+<#import "/templates/links.adoc" as links>
+
+<@tmpl.guide
+title="Configuring Keycloak"
+summary="An overview of the server configuration">
+
+In this guide, you will learn the core concepts around the server configuration and how to configure the server using the different
+server options available. You will also learn how to properly configure the server to achieve an optimal runtime for faster
+startup and low memory footprint.
+
+== Understanding Server Options
+
+All the server options can be found by using the `help` option for individual commands:
+
+<@kc.all parameters="--help"/>
+
+You can instead look at all the server options at <@links.server id="all-config"/>.
+
+Server options are loaded from different sources in a specific order and they use different formats:
```

Comment:
I'd add a sentence about the ordinal concept, which makes clear that CLI takes precedence over Environment takes precedence over config file. I know that the table should show this, but I think it does not get clear enough for people looking at it the first time without prior knowledge.

Reply:
I agree, it's not quite explicit if it's first in the list, or last in the list that takes precendence. I would assume it's first, but could also misinterpret that